runtime/src/manifest.rs

```diff
@@ -57,6 +57,15 @@ fn check_hash(hash: &Option<String>, data: &[u8]) -> Result<(), Error> {
 
 const WASM: &[u8] = include_bytes!("extism-runtime.wasm");
 
+fn new_module(engine: &Engine, buf: &[u8]) -> Result<Module, Error> {
+    let has_magic = buf.len() >= 4 && buf[0..4] == WASM_MAGIC;
+    if has_magic {
+        Ok(Module::new(engine, buf)?)
+    } else {
+        Ok(unsafe { Module::deserialize(engine, buf)? })
```

Comment:
Could be potentially further sped up by leveraging https://docs.wasmtime.dev/api/wasmtime/struct.Module.html#method.from_trusted_file

It implicitly checks the magic bytes as well and has this same if/else. The major difference is it mmaps the file and some other stuff, reportedly making it faster.

Reply:
I had considered that, but we already have the file in memory at this point, so reading the file a second time feels a bit wasteful? Maybe there is some additional refactoring that could make this possible.